src/grpc/clients/ea_restaurant_java_etl_grpc_client.py

```diff
@@ -47,7 +60,70 @@ def insert_mongo_order_status_histories_from_python_etl(
                 mongoOrderStatusHistory=mongo_order_status_histories
             )
         )
-        metadata = (("authorization", access_token),)
+        metadata = (
+            ("authorization", generate_bearer_token_from_access_token(access_token)),
+        )
         return self.mongo_order_status_service_stub.insertMongoOrderStatusHistoriesFromPythonEtl(
             mongo_order_status_histories_request, metadata=metadata
         )
+
+    def execute_service_with_oauth2_auth(self, grpc_call):
```

Comment:
I think this can be private. I would recomend to expose only the calls provided by the server for example `insert_mongo_order_status_histories_from_python_etl`.

The implementation should look like this:

```
    def insert_mongo_order_status_histories_from_python_etl(
        self, mongo_order_status_histories, access_token
    ):
        mongo_order_status_histories = (
            map_mongo_order_status_histories_to_grpc_mongo_order_status_histories(
                mongo_order_status_histories
            )
        )
        mongo_order_status_histories_request = (
            java_etl_grpc_client_pb2.MongoOrderStatusHistoriesFromPythonRequest(
                mon
goOrderStatusHistory=mongo_order_status_histories
            )
        )

       def run_grpc_call(metadata):
        return self.mongo_order_status_service_stub.insertMongoOrderStatusHistoriesFromPythonEtl(
            mongo_order_status_histories_request, metadata=metadata
        )

       response = run_with_oauth2(run_grpc_call)

        return response.mongo_uuids
```

then for `run_with_oauth2`:

```
def run_with_oauth2(self, grpc_call, token_refreshed = False, last_try = False):
        if not self.access_token:
           self.login(self.client_id, self.client_secret)

        metadata = (("authorization", self.access_token),)

        try:
            return grpc_call(metadata)

        except Exception as e:
            if (e.code().value[0] != StatusCode.PERMISSION_DENIED.value or last_try):
               raise grpc_call_exception
          
           if token_refreshed:
              self.access_token = None
              return run_with_oauth2(grpc_call, False, True)
              

            self._refresh_token(self.client_id, self.client_secret, self.acess_token)
            return run_with_oauth2(grpc_call, True)
```

After this I think you can remove:

* `send_mongo_order_status_history_with_oauth2_auth`
* `_login_client`
* `_refresh_token`